### ISSUE TITLE
[Bug] Refused to connect to '...' because it violates the following Content Security Policy directive: "connect-src 'self'".

### ISSUE THREAD

**Issue by BlipRanger** [iv-org/invidious#3665]
**Describe the bug**

After updating to the latest (0995e04) version of Invidious, I began to be unable to play DASH videos any longer due to the error mentioned in the title: 
```
Refused to connect to 'http://my.domain.com/videoplayback?expire=1677545251&ei=w_r8Y9S0OYfi8wTJtrDgCQ&ip=75.176.24.54&id=o-AKV8UnCzVCcH_7bZ63TeLOYJdkYbEskoVsBWVh7ouuMK&itag=135&source=youtube&requiressl=yes&mh=aY&mm=31%2C26&mn=sn-p5qs7nzr%2Csn-5ualdnll&ms=au%2Conr&mv=m&mvi=1&pl=15&ctier=SH&initcwndbps=1326250&spc=H3gIhumADC0qE-zyX0L9GCPPwyuB_tY&vprv=1&svpuc=1&mime=video%2Fmp4&gir=yes&clen=1317109&dur=43.000&lmt=1677454316794978&mt=1677523273&fvip=3&keepalive=yes&fexp=24007246&c=ANDROID&txp=631A224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cctier%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgaoGcOpu7OY5Qvv9BuWuZVa2rbi9qs6rMFb6q6UiHxWECIQD3gtH1d1XSxa5-0Yvaob3vjLrfY_byEPFpgE0MjDlpcA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhAN5UE-RQUk_PfWH83aF6EXqseX_B-ne8Fu0Rcvh8CmZ1AiEAmOdU_Sd0xdDucvnGzb3UAK604Kk53KncCrMGWhTMCjU%3D&host=rr1---sn-p5qs7nzr.googlevideo.com' because it violates the following Content Security Policy directive: "connect-src 'self'". 
```
This error subsequently throws ``` VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) The media could not be loaded, either because the server or network failed or because the format is not supported. ``` and nothing is played or loaded. 

**Steps to Reproduce**
1. Update to the latest version.
2. Attempt to play a video using DASH.
-->

**Logs**
No applicable logs visible beyond the typical request info. 

**Screenshots**
N/A

**Additional context**
 - Browser (if applicable): Tested on both Chromium and FF browsers.
 - OS (if applicable): Windows
 - Docker based install of Invidious

**Comment by BlipRanger** [comment]
> Does it work on older commits though?

It had, I'm not sure how far back I was before pulling this update though. Likely a month back. I'm probably going to roll back to before I made the update, but wanted to see if I could resolve the issue before doing so.

**Comment by unixfox** [comment]
It does seem like a server issue on your side though because we didn't change anything related to DASH and the latest version is working fine for everyone using it.

**Comment by BlipRanger** [comment]
> It does seem like a server issue on your side though because we didn't change anything related to DASH and the latest version is working fine for everyone using it.

Right, fair enough. I've replicated the issue on different devices and browsers, but can't seem to figure out what might be causing this. I'll likely just roll back then and see if that resolves this. Related, is there any easy way to backup the watch history? I think that's the only thing I'd lose by rolling back.

**Comment by unixfox** [comment]
You don't loose anything by using an older docker tag.

**Comment by BlipRanger** [comment]
> You don't loose anything by using an older docker tag.

Yeah, rolling back to "05258d56bdc3f4de1f0da0c0dbd2d540f68cbdd5" made DASH work for me again. I still wonder why that might happen.

**Comment by BlipRanger** [comment]
Reopening because I did a full install from scratch (Docker) and I'm still running into the issue. All DASH video seem to do this. I'm currently running behind Traefik with the following headers:
```
      - "traefik.http.middlewares.invidious-header.headers.referrerPolicy=no-referrer"
      - "traefik.http.middlewares.invidious-header.headers.stsSeconds=31536000"
      - "traefik.http.middlewares.invidious-header.headers.forceSTSHeader=true"
      - "traefik.http.middlewares.invidious-header.headers.stsPreload=true"
      - "traefik.http.middlewares.invidious-header.headers.stsIncludeSubdomains=true"
      - "traefik.http.middlewares.invidious-header.headers.browserXssFilter=true"
      - "traefik.http.middlewares.invidious-header.headers.customRequestHeaders.X-Forwarded-Proto=https"
 ```
 Is there perhaps something missing in the reverse proxying that is now necessary? This issue was not occurring with these proxy settings for versions 05258d56bdc3f4de1f0da0c0dbd2d540f68cbdd5 and before.

**Comment by BlipRanger** [comment]
Another clue, a page I had with a video open from before I did the reinstall was able to continue playing without this issue. I changed the resolution and it was able to buffer and play still. So this leads me to believe that it is something intrinsic to the initial request.

**Comment by unixfox** [comment]
What are the HTTP headers sent by invidious? Not by traefik.

**Comment by BlipRanger** [comment]
> What are the HTTP headers sent by invidious? Not by traefik.

```
content-length: 7318
content-security-policy: default-src 'none'; script-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:; connect-src 'self'; manifest-src 'self'; media-src 'self' blob: https://*.googlevideo.com:443 https://*.youtube.com:443; child-src 'self' blob:; frame-src 'self'; frame-ancestors 'none'
content-type: text/html
date: Wed, 15 Mar 2023 19:29:24 GMT
permissions-policy: interest-cohort=()
referrer-policy: same-origin
x-content-type-options: nosniff
x-frame-options: sameorigin
x-xss-protection: 1; mode=block
```

**Comment by unixfox** [comment]
are you browsing on the same domain that you configured in the config.yml?

**Comment by BlipRanger** [comment]
> are you browsing on the same domain that you configured in the config.yml?

Yep, it's been the same for years. Very little changed in my docker-compose beyond the version (and the few new config lines that were added in the env).

**Comment by BlipRanger** [comment]
I think I've figured it out. 
Setting `https_only: true` seems to have resolved this issue for me. I'm not sure if it was changed at some point in my compose file  (I'm pretty sure it didn't)  or something changed in the way the CSP header was being created and I just hadn't updated since then. 

The clue was that the `Refused to connect to 'http://my.domain.com/` features the _http_ url, not the _https_ one, which I assume is what the CSP treats as self.
